src/services/TestResultsService.ts

```diff
@@ -244,117 +249,117 @@ export class TestResultsService {
     if (payload.testTypes.length > 0) {
       payload.testTypes.forEach((testType: any) => {
         Object.assign(testType,
-          {
-            createdAt: new Date().toISOString(), lastUpdatedAt: new Date().toISOString()
-          });
+            {
+              createdAt: new Date().toISOString(), lastUpdatedAt: new Date().toISOString()
+            });
       });
     }
     return payload;
   }
 
-  public setExpiryDateAndCertificateNumber(payload: ITestResultPayload) {
-    if (payload.testStatus !== "submitted") {
+  public generateExpiryDate(payload: ITestResultPayload) {
+    if (payload.testStatus !== TEST_STATUS.SUBMITTED) {
       return Promise.resolve(payload);
     } else {
       return this.getMostRecentExpiryDateOnAllTestTypesByVin(payload.vin)
-        .then((mostRecentExpiryDateOnAllTestTypesByVin) => {
-          payload.testTypes.forEach((testType: any, index: number) => {
-            if (testType.testTypeClassification === "Annual With Certificate" &&
-              (testType.testResult === "pass" || testType.testResult === "prs" || testType.testResult === "fail")) {
-              testType.certificateNumber = testType.testNumber;
-              payload.testTypes[index] = testType;
-              if (testType.testResult !== "fail") {
```

Comment:
this is changed because now this function only generates expiryDate and not certificateNumber anymore. because the expiryDate needs to be generated only for pass and prs, hence the change. hopefully I answered your question